File: backup_before_industrialisation_20250626/blueprints/tests_centralises/api/routes/test_asset_routes.py
"""
Test ultra avancé pour le blueprint asset_routes (Flask)
"""
from flask import Flask
from blueprints.api.routes.asset_routes import bp

def test_asset_routes():
    app = Flask(__name__)
    app.register_blueprint(bp)
    client = app.test_client()
    resp = client.get('/assets')
    assert resp.status_code == 200
    assert resp.json[0]["name"] == "Asset 1"
    resp = client.post('/assets', json={"name": "Asset 2"})
    assert resp.status_code == 200
    assert resp.json["message"] == "Asset créé"

File: blueprints/tests_centralises/api/routes/test_asset_routes.js
// Test ultra avancé pour le blueprint asset_routes (Express)
const request = require('supertest');
const express = require('express');
const assetRoutes = require('../../../../blueprints/api/routes/asset_routes');

describe('assetRoutes Express', () => {
  let app;
  beforeAll(() => {
    app = express();
    app.use(express.json());
    app.use(assetRoutes);
  });
  it('GET /assets doit répondre 200', async () => {
    const res = await request(app).get('/assets');
    expect(res.statusCode).toBe(200);
    expect(Array.isArray(res.body)).toBe(true);
  });
  it('POST /assets doit répondre 200', async () => {
    const res = await request(app).post('/assets').send({ name: 'Asset 2' });
    expect(res.statusCode).toBe(200);
    expect(res.body.message).toMatch(/créé/);
  });
});
